### PR TITLE
Profile settings

### PR DESCRIPTION
**Pull request by broskoski**
Re: https://github.com/aredotna/ervell/issues/1198

profile-settings

### CODE REVIEW

File: apps/settings/components/form_card/index.coffee [hunk 1]
```diff
@@ -12,15 +12,22 @@ module.exports = ($el) ->
...
-      method: $el.data('method')
-      data: serializer.data()
-
+    Promise.all [
+      $.ajax(
+        url: $el.data('action')
+        method: $el.data('method')
+        data: serializer.data()
+      ),
+      onSubmit()
```
Comment (dzucconi) [inline apps/settings/components/form_card/index.coffee]: What is this onSubmit?

File: apps/settings/components/form_card/index.coffee [hunk 2]
```diff
@@ -12,15 +12,22 @@ module.exports = ($el) ->
 
     serializer = new Serializer $el
 
+    onSubmit = if onSubmit then onSubmit else ->
+      promise = new Promise
+      promise.resolve()
```
Comment (dzucconi) [inline apps/settings/components/form_card/index.coffee]: I mean I see it here obviously but I don't understand what it could be doing.

Comment (broskoski) [reply apps/settings/components/form_card/index.coffee]: https://github.com/aredotna/ervell/pull/1246/files#diff-5a9d3936c693b3f81620b002aa5b99c5R14

File: apps/settings/components/form_card/index.coffee (hunk 1 above)
Comment (dzucconi) [reply apps/settings/components/form_card/index.coffee]: Oh you're trying to optionally include a promise returning fn.

Comment (dzucconi) [reply apps/settings/components/form_card/index.coffee]: Two things: this is fine but very awkward, that part doesn't matter much. But $.ajax doesn't return an actual promise. You need to wrap it in Promise

File: apps/settings/client/tabs/general.coffee [hunk 3]
```diff
@@ -9,7 +10,16 @@ module.exports = ($el) ->
 
   models = user: new User(USER)
 
-  $el.find('form').each (_i, form) -> formCard $(form)
+  onSubmit = ->
+    currentMeta = models.user.get('metadata')
+    currentMeta = {} if not currentMeta
+    currentMeta.description = $el.find('.js-bio').val()
+
+    $.post "#{models.user.urlRoot()}/metadata",
```
Comment (dzucconi) [inline apps/settings/client/tabs/general.coffee]: Not a promise

Comment (dzucconi) [reply apps/settings/client/tabs/general.coffee]: `Promise $.post ...`

File: apps/settings/components/form_card/index.coffee (hunk 2 above)
Comment (dzucconi) [reply apps/settings/components/form_card/index.coffee]: `Promise.resolve`

File: apps/settings/components/form_card/index.coffee [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Promise = require 'bluebird-q'
 Serializer = require '../../../../components/form/serializer.coffee'
 
-module.exports = ($el) ->
+module.exports = ($el, onSubmit) ->
```
Comment (dzucconi) [inline apps/settings/components/form_card/index.coffee]: `onSubmit = Promise.resolve()`

File: apps/settings/components/form_card/index.coffee [hunk 5]
```diff
@@ -16,11 +16,14 @@ module.exports = ($el) ->
       .prop 'disabled', true
       .text 'Saving...'
 
-    Promise $.ajax
-      url: $el.data('action')
-      method: $el.data('method')
-      data: serializer.data()
-
+    Promise.all [
+      $.ajax(
```
Comment (dzucconi) [inline apps/settings/components/form_card/index.coffee]: `Promise $.ajax`

**Pull request merged**